### PR TITLE
test(coverage): expand apme_engine unit coverage and gate at 70

### PR DESCRIPTION
**Pull request by djdanielsson**
## Summary

Unit coverage expansion for `src/apme_engine` plus a ratcheted gate. Split out of #646 for independent review — this branch contains **only** the four new test modules plus gate/docs changes (7 files), based directly on `main`.

## Changes

- `tests/test_daemon_entrypoints.py` — daemon mains, validator gRPC bootstrap, daemon CLI, health checks
- `tests/test_engine_graph_coverage.py` — engine models, finder, parser, utils
- `tests/test_engine_pipeline_coverage.py` — scanner, loader, yaml, keyutil
- `tests/test_daemon_runtime_coverage.py` — launcher, health, chunked FS, sinks, validators
- `tox.ini` / `pyproject.toml`: coverage floor aligned 36/50 → 70 (measured 77% on this branch)
- `DEVELOPMENT.md`: gate docs, `--no-cov` filtered-run note, engine-only scope note

The final ratchet to 80 follows once #647/#648/#650 land with their tests (combined suite measures 84%).

### Review feedback (55bfa166)

- Patched `_MARKER_FILE` in `test_launcher_save_load_roundtrip` to avoid deleting real daemon markers
- Replaced tautological assertions (`or True`) with real contract checks (module sorting, galaxy proxy URL)
- Parameterized key-setter test with correct object types per setter
- Removed ineffective `get_event_loop` failure injection in ansible validator test
- Fixed `test_chunked_should_include_stat_oserror` CI failure (Python 3.12+ `is_file` OSError propagation)

## Test plan

- [x] `tox -e lint` passes
- [x] `tox -e unit` passes (3785 passed, 49 skipped, 77% coverage, gate enforced)
- [x] Docs updated
- [ ] ADR added (not applicable)

## Summary by CodeRabbit

- **Tests**
  - Expanded automated coverage for daemon operations, health checks, service lifecycles, engine processing, error handling, and output formats.
  - Added tests for edge cases, retries, cleanup, malformed inputs, and fallback behavior.

- **Documentation**
  - Updated coverage guidance to reflect the 70% target, enforced through continuous integration checks.
  - Clarified measurement scope, excluded areas, and future coverage goals.

- **Chores**
  - Increased the minimum unit-test coverage requirement to 70%.
  - Updated test environments to use the database URL configuration consistently.

### CODE REVIEW

File: tests/test_daemon_runtime_coverage.py [hunk 1]
```diff
@@ -0,0 +1,2516 @@
...
+
+    Args:
+        tmp_path: Pytest temporary directory.
+        monkeypatch: Pytest monkeypatch fixture.
+    """
+    from apme_engine.daemon import launcher
+
+    data_dir = tmp_path / "data"
+    monkeypatch.setattr(launcher, "_DATA_DIR", data_dir)
+    monkeypatch.setattr(launcher, "_STATE_FILE", data_dir / "daemon.json")
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply tests/test_daemon_runtime_coverage.py:2198]: Patched `_MARKER_FILE` to the test temp directory alongside `_DATA_DIR` and `_STATE_FILE` so `DaemonState.remove()` only touches test-controlled files. Fixed in 55bfa166.

File: tests/test_engine_graph_coverage.py [hunk 2]
```diff
@@ -0,0 +1,2622 @@
...
+        c = Collection(name="ns.col")
+        m2 = Module(name="z", key="module z")
+        m1 = Module(name="a", key="module a")
+        c.modules = [m2, m1, "module m0"]
+        c.playbooks = ["pb2", "pb1"]
+        c.roles = ["r2", "r1"]
+        c.taskfiles = ["t2", "t1"]
+        out = c.children_to_key()
+        assert out is c
+        assert list(c.modules)[:2] == sorted(cast(list[str], list(c.modules)[:2])) or True
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply tests/test_engine_graph_coverage.py]: Replaced the tautological `or True` assertion with exact expected sorted module, role, and taskfile key sequences after `children_to_key()`. Fixed in 55bfa166.

File: tests/test_engine_pipeline_coverage.py [hunk 3]
```diff
@@ -0,0 +1,1706 @@
...
+        t.index = 2
+        keyutil.set_task_key(t, parent_key="taskfile tf#/x", parent_local_key="taskfile tf#/y")
+        assert t.key.startswith("task ")
+        assert "[2]" in t.key
+
+    def test_set_taskfile_playbook_file_keys(self) -> None:
+        """Taskfile/playbook/file keys share the prefix helper."""
+        for setter in [keyutil.set_taskfile_key, keyutil.set_playbook_key, keyutil.set_file_key]:
+            obj = _dummy_key_obj(type="taskfile", collection="ns.col", role="", defined_in="tasks/a.yml")
+            setter(obj)
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply tests/test_engine_pipeline_coverage.py]: Parameterized the test with matching object types (`taskfile`, `playbook`, `file`) and expected local-key prefixes for each setter. Fixed in 55bfa166.

File: tests/test_daemon_runtime_coverage.py [hunk 4]
```diff
@@ -0,0 +1,2516 @@
...
+        "dep_audit": "127.0.0.1:50059",
+        "galaxy_proxy": "127.0.0.1:8765",
+    }
+    old_env = dict(os.environ)
+    try:
+        asyncio.run(launcher._run_daemon(services))
+    finally:
+        os.environ.clear()
+        os.environ.update(old_env)
+    assert os.environ.get("APME_GALAXY_PROXY_URL") is None or True
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply tests/test_daemon_runtime_coverage.py]: Captured `APME_GALAXY_PROXY_URL` before environment restoration and assert it equals `http://127.0.0.1:8765`. Removed the tautological post-restore check. Fixed in 55bfa166.

File: tests/test_daemon_runtime_coverage.py [hunk 5]
```diff
@@ -0,0 +1,2516 @@
...
+            request_id="exc-1",
+            files=[common_pb2.File(path="a.yml", content=b"x\n")],
+            hierarchy_payload=b"{}",
+        )
+        with (
+            patch.object(mod, "_run_ansible_validate", side_effect=RuntimeError("boom")),
+            patch("apme_engine.daemon.ansible_validator_server.asyncio.get_event_loop") as loop_mock,
+        ):
+            loop_mock.side_effect = RuntimeError("no loop")
+            resp = await AnsibleValidatorServicer().Validate(req, MagicMock())
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply tests/test_daemon_runtime_coverage.py]: Removed the `get_event_loop` failure injection; the test now relies solely on `_run_ansible_validate` raising `RuntimeError("boom")` to exercise the executor-error branch. Fixed in 55bfa166.

File: tests/test_daemon_runtime_coverage.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (djdanielsson) [reply tests/test_daemon_runtime_coverage.py:2198]: Patched `_MARKER_FILE` to `data_dir / "daemon.marker"` in `test_launcher_save_load_roundtrip` so cleanup cannot delete a real user daemon marker. Fixed in 55bfa166.

**Comment by djdanielsson** [comment]
@coderabbitai full review
